packages/core/src/components/non-ideal-state/nonIdealState.tsx

```diff
@@ -56,6 +56,13 @@ export interface NonIdealStateProps extends Props {
      */
     iconSize?: NonIdealStateIconSize;
 
+    /**
+     * Whether the `icon` should attempt to use a muted style.
+     *
+     * @default true
+     */
+    mutedIcon?: boolean;
```

Comment:
nit: let's call this `iconMuted` to get the prop visually grouped with the other icon-related props in alphabetical sorting

```suggestion
    iconMuted?: boolean;
```